src/devices/device_constructors/hydrogeneration_constructor.jl

```diff
@@ -33,6 +33,88 @@ function construct_device!(canonical::Canonical, sys::PSY.System,
 
 end
 
+function construct_device!(canonical::Canonical, sys::PSY.System,
+                           model::DeviceModel{H, D},
+                           ::Type{S};
+                           kwargs...) where {H<:PSY.HydroGen,
+                                             D<:HydroDispatchSeasonalFlow,
```

Comment:
you don't need to do D<:HydroDispatchSeasonalFlow since this is a concrete type or you might do it for the Abstract.